### PR TITLE
Import `basic_paradigm` instead of having multiple definitions

### LINKED ISSUE #2763: Multiple definitions of `basic_paradigm`
The function `basic_paradigm` is defined in `nilearn._utils.data_gen.py`:

https://github.com/nilearn/nilearn/blob/00d3873fb6cf23d09f8eba6d0d20e905f26d22dd/nilearn/_utils/data_gen.py#L505-L510

But, it is re-defined in other files (with a duration) for testing:

- `nilearn/glm/tests/test_first_level.py`:
https://github.com/nilearn/nilearn/blob/a36c0442fc8eebe401b310dd9799c0f071080e90/nilearn/glm/tests/test_first_level.py#L353-L360

- `nilearn/glm/tests/test_paradigm.py`:
https://github.com/nilearn/nilearn/blob/a36c0442fc8eebe401b310dd9799c0f071080e90/nilearn/glm/tests/test_paradigm.py#L16-L23

- `nilearn/glm/tests/test_dmtx.py`:
https://github.com/nilearn/nilearn/blob/a36c0442fc8eebe401b310dd9799c0f071080e90/nilearn/glm/tests/test_dmtx.py#L48-L55

It would be better to remove those and import `basic_paradigm` from `_utils.data_gen.py`.

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #2763. 
Replaces and closes #2770.

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: nilearn/glm/tests/test_first_level.py [hunk 1]
```diff
@@ -14,7 +14,8 @@
 
 from nilearn._utils.data_gen import (create_fake_bids_dataset,
                                      generate_fake_fmri_data_and_design,
-                                     write_fake_fmri_data_and_design)
+                                     write_fake_fmri_data_and_design,
+                                     basic_paradigm)
```
Comment (NicolasGensollen) [inline nilearn/glm/tests/test_first_level.py:18]: You need to handle the definition of `basic_paradigm` line 475 in this file.

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @ymzayek !
LGTM minus the small issue below.

File: nilearn/glm/tests/test_first_level.py (hunk 1 above)
Comment (ymzayek) [reply nilearn/glm/tests/test_first_level.py:18]: Oops missed that one, thanks!

**Review by tsalo — CHANGES_REQUESTED** [review]
I just have a couple of suggestions. Please feel free to ignore the proposed change to the `basic_paradigm` code, but I think it might be best to add a docstring in this PR, rather than #3262.

File: nilearn/_utils/data_gen.py [hunk 2]
```diff
@@ -501,11 +501,17 @@ def generate_group_sparse_gaussian_graphs(
...
 
 
-def basic_paradigm():
-    conditions = ['c0', 'c0', 'c0', 'c1', 'c1', 'c1', 'c2', 'c2', 'c2']
+def basic_paradigm(condition_names_have_spaces=False):
+    if condition_names_have_spaces:
+        conditions = ['c 0', 'c 0', 'c 0', 'c 1', 'c 1', 'c 1',
+                      'c 2', 'c 2', 'c 2']
+    else:
+        conditions = ['c0', 'c0', 'c0', 'c1', 'c1', 'c1', 'c2', 'c2', 'c2']
```
[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/_utils/data_gen.py:509]: Yes, that looks better thanks!

File: nilearn/_utils/data_gen.py [hunk 3]
```diff
@@ -501,11 +501,17 @@ def generate_group_sparse_gaussian_graphs(
     return signals, precisions, topology
 
 
-def basic_paradigm():
-    conditions = ['c0', 'c0', 'c0', 'c1', 'c1', 'c1', 'c2', 'c2', 'c2']
+def basic_paradigm(condition_names_have_spaces=False):
```
[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/_utils/data_gen.py:504]: Yes good catch, I will add it here instead.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @ymzayek ! 👍

**Comment by ymzayek** [comment]
Great, merging!

**Pull request merged**